src/tests/serde.rs

```diff
@@ -594,3 +600,188 @@ fn test_serialize_deserialize_unsigned_numbers() {
+    let b = B {
+        num_1: 1,
+        num_2: i64::MAX as u64 + 1,
+    };
+    let doc_result = to_document(&b);
+    assert!(doc_result.is_err());
+}
+
+#[test]
+fn test_datetime_helpers() {
+    let _guard = LOCK.run_concurrently();
+
+    #[derive(Deserialize, Serialize)]
+    struct A {
+        #[serde(deserialize_with = "serde_helpers::deserialize_bson_datetime_from_ext_json")]
+        pub date: DateTime,
+    }
+
+    let date = DateTime(chrono::DateTime::from_str("2020-06-09 10:58:07.095 UTC").unwrap());
+    let a = A { date };
+    let bson = to_bson(&a).unwrap();
+    let value: Value = bson.into();
+    let a: A = serde_json::from_value(value).unwrap();
+    assert_eq!(a.date, date);
+
+    #[derive(Deserialize)]
+    struct B {
+        #[serde(deserialize_with = "serde_helpers::deserialize_chrono_datetime_from_ext_json")]
+        pub date: chrono::DateTime<chrono::Utc>,
+    }
+
+    let date = r#"
+    {
+        "date": {
+                "$date": {
+                    "$numberLong": "1591700287095"
+                }
+        }
+    }"#;
+    let json: Value = serde_json::from_str(&date).unwrap();
+    let b: B = serde_json::from_value(json).unwrap();
+    let expected: chrono::DateTime<chrono::Utc> =
+        chrono::DateTime::from_str("2020-06-09 10:58:07.095 UTC").unwrap();
+    assert_eq!(b.date, expected);
+
+    #[derive(Deserialize, Serialize)]
+    struct C {
+        #[serde(
+            serialize_with = "serde_helpers::serialize_iso_string_as_datetime",
+            deserialize_with = "serde_helpers::deserialize_iso_string_from_datetime"
+        )]
+        pub date: String,
+    }
+
+    let date = "2020-06-09 10:58:07.095 UTC";
+    let c = C {
+        date: date.to_string(),
+    };
+    let doc = to_document(&c).unwrap();
+    assert!(doc.get_datetime("date").is_ok());
+    let c: C = from_document(doc).unwrap();
+    assert_eq!(c.date.as_str(), date);
+}
+
+#[test]
+fn test_oid_helpers() {
+    let _guard = LOCK.run_concurrently();
+
+    #[derive(Deserialize, Serialize)]
+    struct A {
+        #[serde(deserialize_with = "serde_helpers::deserialize_object_id_from_ext_json")]
+        oid: ObjectId,
+    }
+
+    let oid = ObjectId::new();
+    let a = A { oid: oid.clone() };
+    let bson = to_bson(&a).unwrap();
+    let value: Value = bson.into();
+    let a: A = serde_json::from_value(value).unwrap();
+    assert_eq!(a.oid, oid);
+
+    #[derive(Serialize)]
+    struct B {
+        #[serde(serialize_with = "serde_helpers::serialize_hex_string_as_object_id")]
+        oid: String,
+    }
+
+    let oid = ObjectId::new();
+    let b = B {
+        oid: oid.to_string(),
+    };
+    let doc = to_document(&b).unwrap();
+    assert_eq!(doc.get_object_id("oid").unwrap(), &oid);
+}
+
+#[test]
+fn test_uuid_helpers() {
+    let _guard = LOCK.run_concurrently();
+
+    #[derive(Serialize, Deserialize)]
+    struct A {
+        #[serde(
+            serialize_with = "serde_helpers::serialize_uuid_as_binary",
```

Comment:
Ditto above about considering a module for allowing roundtripping with a single attribute.